components/loading/LoadingBar.vue

```diff
@@ -0,0 +1,59 @@
+<template>
+  <v-app ref="loaderContainer" class="v-app">
+    <v-progress-linear
+      :active="loading"
+      :indeterminate="indeterminate"
+      :height="4"
+      :bottom="true"
+      :rounded="true"
+      :color="barColor"
+      :value="barValue"
+    ></v-progress-linear>
+  </v-app>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+
+let timeout: number;
+export default Vue.extend({
+  data() {
+    return {
+      loading: false,
+      barColor: 'primary',
+      indeterminate: true,
+      barValue: 0
+    };
+  },
+  methods: {
+    start() {
+      window.clearTimeout(timeout);
+      this.loading = true;
+      this.indeterminate = true;
+      this.barColor = 'primary';
+      this.barValue = 0;
+    },
+    finish() {
+      this.indeterminate = true;
+      this.barColor = 'primary';
+      this.barValue = 0;
+      // Ensure that the 'slide up' animation appears after loading finishes. Without this
+      // the animation could be skipped due to DOM blocking.
+      timeout = window.setTimeout(() => (this.loading = false), 50);
```

Comment:
In addition to hiding the bar, the timeout method should also reset the data to its original state, imo. This allows you to avoid resetting everything in every state.

Moving it to a method would also probably be better.